src/main/java/slaybot/TaskList.java

```diff
@@ -0,0 +1,99 @@
+package slaybot;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import entity.Task;
+
+/**
+ * The TaskList class represents a collection of tasks in the SlayBot application.
+ * It provides methods to manipulate and interact with the list of tasks, such as adding,
+ * removing, marking, and searching for tasks.
+ */
+public class TaskList {
+    private List<Task> tasks;
+
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    public List<Task> getTasks() {
+        return this.tasks;
+    }
+
+    public int getSize() {
+        return this.tasks.size();
+    }
+
+    public void removeTask(int index) {
+        this.tasks.remove(index);
+    }
+
+    /**
+     * Adds a task to the TaskList.
+     *
+     * @param t The task to be added.
+     */
+    public void addTask(Task t) {
+        this.tasks.add(t);
+    }
+
+    /**
```

Comment:
Unused code could be deleted for a more elegant codebase :)

Reply:
Will clean up my code as well as add more JavaDocs, thank you! @c-wenlong